json_serializable/lib/src/type_helpers/json_helper.dart

```diff
@@ -273,17 +273,21 @@ ClassConfig? _annotation(ClassConfig config, InterfaceType source) {
     return null;
   }
   final annotations = const TypeChecker.fromRuntime(JsonSerializable)
-      .annotationsOfExact(source.element, throwOnUnresolved: false)
+      .annotationsOfExact(source.element2, throwOnUnresolved: false)
       .toList();
 
   if (annotations.isEmpty) {
     return null;
   }
 
+  if (source.element2 is! ClassElement) {
+    return null;
+  }
```

Comment:
tbh, I'm not 100% sure here.  

`source.element` was a `ClassElement`  but `source.element2` is an `InterfaceElement`. 

I did this type check to ensure only `ClassElement` go through since `mergeConfig` only accepts that type.